File: TestLogic/Interface/source/IfLogicStub.cpp
#include "../include/IfLogicStub.h"
#include "Public/include/TypeMarshaller.h"
#include "../include/IfLogicMethodId.h"
namespace test_logic 
{
TInt32 IfLogicStub::Test(TUInt64 _uParm1,TInt32 _n2)
{
    TInt32 nLen = GetLength(_uParm1)+GetLength(_n2);
    CMessageHeader *pMsg = m_pOnwerObj->PrepareMsg(nLen,(TEST_TUINT64_TINT32_ID),&m_tTarget,1,false);
    if (NULL == pMsg)
    {
        return OUT_OF_MEM;
    }
    TUInt32 nUsed=0;
    TInt32 nRet=0;
    TUChar *pBuffer = pMsg->GetBody();
    nRet = Marshall(pBuffer+nUsed,nLen,_uParm1);
    if (nRet < SUCCESS)
    {
        return nRet;
    }
    nUsed += nRet;
    nLen-=nRet;
    nRet = Marshall(pBuffer+nUsed,nLen,_n2);
    if (nRet < SUCCESS)
    {
        return nRet;
    }
    nUsed += nRet;
    nLen-=nRet;
    pMsg->ResetBodyLength(nUsed);
    return m_pOnwerObj->SendMsg(pMsg);
}

}

File: TestLogic/Interface/IfLogicStub.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using System;
public class IfLogicStub : IfLogic
{
    int IfLogic.Test(ulong _uParm1,int _n2)
    {
        int nLength =TypeMarshaller.GetLength( _uParm1)+TypeMarshaller.GetLength( _n2);
        CMessage pMsg = new CMessage((uint)IfLogicMethodId.IfLogicTestMethodId, nLength);
        if (null == pMsg.m_pBuffers)
        {
            return MacrosAndDef.OUT_OF_MEM;
        }
        int nBufferLen = pMsg.m_pBuffers.Length;
        int nUsed = TypeMarshaller.Marshall(pMsg.m_pBuffers, nBufferLen, 0, nLength);
        nUsed += TypeMarshaller.Marshall(pMsg.m_pBuffers, nBufferLen, nUsed,pMsg.m_uMsgId);
        int nLen = TypeMarshaller.Marshall(pMsg.m_pBuffers, nBufferLen, nUsed, _uParm1);
        if (nLen < MacrosAndDef.SUCCESS)
        {
            return nLen;
        }
        nUsed += nLen;
        nLen = TypeMarshaller.Marshall(pMsg.m_pBuffers, nBufferLen, nUsed, _n2);
        if (nLen < MacrosAndDef.SUCCESS)
        {
            return nLen;
        }
        nUsed += nLen;
        if (nUsed != nBufferLen)
        {
            Debug.Log("the nUsed is:"+nUsed);
            TypeMarshaller.Marshall(pMsg.m_pBuffers, nBufferLen, sizeof(uint), nUsed);
        }
        CMessage.sm_pSocket.SendMsg(pMsg);
        return nUsed;
    }
}
